src/tests/ftest/security/cont_delete_acl.py

```diff
@@ -0,0 +1,141 @@
+    # pylint: disable=too-many-ancestors
+    """Test Class Description:
+
+    Test to verify ACL entry deletion.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up each test case."""
+        super(DeleteContainerACLTest, self).setUp()
+        self.daos_cmd = self.get_daos_command()
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+        # Get list of ACL entries
+        cont_acl = self.get_container_acl_list(
+            self.pool.uuid, self.pool.svc_ranks[0], self.container.uuid)
+
+        # Get principals
+        self.principals_table = {}
+        for entry in cont_acl:
+            self.principals_table[entry.split(":")[2]] = entry
+
+    def test_acl_delete_invalid_inputs(self):
+        """
+        JIRA ID: DAOS-3714
+
+        Test Description: Test that container delete command performs as
+            expected with invalid inputs.
+
+        :avocado: tags=all,pr,security,container_acl,cont_delete_acl
+        """
+        # Get list of invalid ACL principal values
+        invalid_principals = self.params.get("invalid_principals", "/run/*")
+
+        # Check for failure on invalid inputs.
+        for principal in invalid_principals:
+            try:
+                self.daos_cmd.container_delete_acl(
+                    self.pool.uuid,
+                    self.pool.svc_ranks[0],
+                    self.container.uuid,
+                    principal)
+            except process.CmdError as err:
+                if "-1003" in err.result.stderr:
+                    self.log.info(
+                        "Found expected error %s with invalid principal %s",
+                        err.result.stderr,
+                        principal)
+                else:
+                    self.fail("delete-acl seems to have failed with \
+                        unexpected error: {}".format(err))
+        self.fail("container delete-acl command expected to fail.")
+
+    @fail_on(CommandFailure)
+    def test_delete_valid_acl(self):
+        """
+        JIRA ID: DAOS-3714
+
+        Test Description: Test that container delete command successfully
+            removes principal in ACL.
+
+        :avocado: tags=all,pr,security,container_acl,cont_delete_acl
+        """
+        for principal in self.principals_table:
+            self.daos_cmd.container_delete_acl(
+                self.pool.uuid,
+                self.pool.svc_ranks[0],
+                self.container.uuid,
+                principal)
+            if self.principals_table[principal] in self.daos_cmd.result.stdout:
+                self.fail(
+                    "Found acl that was to be deleted in output: {}".format(
+                        self.daos_cmd.result.stdout))
+
+    def test_no_user_permissions(self):
+        """
+        JIRA ID: DAOS-3714
+
+        Test Description: Test that container delete command successfully
+            removes principal in ACL.
```

Comment:
```suggestion
        Test Description: Test that container delete command doesn't
            remove principal in ACL without permission.
```
Along those lines...

Reply:
Fixed.